src/context/network.js

```diff
@@ -84,14 +86,30 @@ function networkReducer(state, action) {
     case ADD_CONNECTION: {
         const { interfaces, connections } = state;
         const conn = action.payload;
+
         const iface = createInterface({ name: conn.name, type: conn.type });
+
         return {
             ...state,
             interfaces: { ...interfaces, [iface.id]: iface },
             connections: { ...connections, [conn.id]: conn }
         };
     }
 
+    case CONFIGURE_INTERFACE: {
```

Comment:
To be honest, I'm not sure about it. If you review it commit per commit you'll see that before introducing this specific action [I modified the `ADD_CONNECTION`](https://github.com/dgdavid/cockpit-wicked/pull/58/commits/b783f09da0bd29b28d62baf39240a03776584595)

What do you think is better?

Reply:
I prefer extending the `ADD_CONNECTION` action, to be honest. This `CONFIGURE_INTERFACE` introduces a level of indirection that simply does not worth.